enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,454 @@
+specialized manifests from the default manifests.
+
+### User Stories
+
+The initial direct user of the single-node-developer cluster profile
+is the SNC image build process, which is currently used by CRC as well
+as some additional projects. If any of these indirect users did not
+need the actual VM image provided by SNC, they could switch to using
+the new cluster profile directly via the standard OpenShift installer.
+
+#### Story 1
+
+At the end of phase one, the SNC image build process will be slightly
+simpler, and should run a bit faster. The resulting CRC cluster that
+end users deploy should be unchanged.
+
+#### Story 2
+
+As phase two progresses with merges to various components, the SNC
+image build process will become even simpler and quicker, and the
+resulting CRC cluster should require less system resources.
+
+### Implementation Details/Notes/Constraints
+
+Phase 1 is pretty straightforward - simply adding an annotation to the
+existing manifest files in the various repositories for components to
+be included in single node developer clusters.
+
+Phase 2 is more of an open ended process leveraging the new profile to
+optimize the configuration of the individual components for single
+node developer clusters. In general, each component will be addressed
+independently and possibly multiple times, first to eliminate the
+post-install patching by SNC, and then to enable additional
+optimizations. During implementation of this phase, it is expected
+that patterns will emerge when common approaches apply across
+different components.
+
+### Risks and Mitigations
+
+One risk is that all parts of the implementation of the cluster
+profiles
+[enhancement](https://github.com/openshift/enhancements/blob/master/enhancements/update/cluster-profiles.md)
+do not get merged in a timely manner. Implementing the
+single-node-developer cluster profile requires the ability to specify
+a non-default cluster profile to the OpenShift installer, and for this
+to result in only the manifests annotated for the specified cluster
+profile to be processed by the resulting cluster. It's not clear at
+this time which team is signed up to implement the cluster profile
+support, but developer tools team members may be available to assist
+in writing code, and definitely will participate in the review
+process.
+
+Another risk is that it will take too long to get the initial
+annotations merged to all the needed OpenShift component repositories
+during the first phase of development. Until all needed annotations
+are merged, the SNC tooling cannot be updated to make use of the
+single-node-developer cluster profile, as needed components would be
+excluded from the cluster. The review process for this enhancement
+should help raise awareness across the various teams that will need to
+review and merge these patches.
+
+A final risk is that new components will be added to OpenShift after
+the completion of phase 1 that would be useful in single node
```

Comment:
I think it’s important to clarify who says it’s useful.

Is the CRC making that choice or each component team?

Reply:
Given the component teams approve patches to their components, they have the ultimate control. But the CRC team would certainly want to collaborate on the decisions, especially if they are not obvious or involve trade-offs. Any suggestions on how to handle this?

Reply:
CRC team understands the goals of CRC and component teams understand their component details. I would expect that both teams would need to collaborate on the CRC goal while the component experts can make sure there is a still a good experience in the smaller single node case...so it still is OpenShift

Reply:
Addressed in various places in update.